src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ioc/ResumePaymentCallbackHandler.java

```diff
@@ -0,0 +1,122 @@
+
+@Service
+public class ResumePaymentCallbackHandler extends CallbackHandler {
+    private static final List<CaseEvent> EVENTS = Collections.singletonList(RESUME_CLAIM_PAYMENT_CITIZEN);
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private final PaymentsService paymentsService;
+    private final CaseDetailsConverter caseDetailsConverter;
+    private final CaseMapper caseMapper;
+    private final IssueDateCalculator issueDateCalculator;
+    private final ResponseDeadlineCalculator responseDeadlineCalculator;
+
+    @Autowired
+    public ResumePaymentCallbackHandler(
+        PaymentsService paymentsService,
+        CaseDetailsConverter caseDetailsConverter,
+        CaseMapper caseMapper,
+        IssueDateCalculator issueDateCalculator,
+        ResponseDeadlineCalculator responseDeadlineCalculator) {
+        this.paymentsService = paymentsService;
+        this.caseDetailsConverter = caseDetailsConverter;
+        this.caseMapper = caseMapper;
+        this.issueDateCalculator = issueDateCalculator;
+        this.responseDeadlineCalculator = responseDeadlineCalculator;
+    }
+
+    @Override
+    protected Map<CallbackType, Callback> callbacks() {
+        return ImmutableMap.of(
+            CallbackType.ABOUT_TO_SUBMIT, this::resumePayment
+        );
+    }
+
+    @Override
+    public List<CaseEvent> handledEvents() {
+        return EVENTS;
+    }
+
+    private CallbackResponse resumePayment(CallbackParams callbackParams)  {
+        CaseDetails caseDetails = callbackParams.getRequest().getCaseDetails();
+        String authorisation = callbackParams.getParams()
+            .get(CallbackParams.Params.BEARER_TOKEN).toString();
+
+        Claim claim = caseDetailsConverter.extractClaim(caseDetails);
+
+        Claim claimAfterPayment = withPayment(authorisation, claim, caseDetails.getId());
+
+        return AboutToStartOrSubmitCallbackResponse
+            .builder()
+            .data(caseDetailsConverter.convertToMap(caseMapper.to(claimAfterPayment)))
+            .build();
+    }
+
+    private Claim withPayment(String authorisation, Claim claim, Long caseId) {
+        Payment originalPayment = paymentsService.retrievePayment(authorisation, claim);
+
+        if (originalPayment.getStatus().equals(INITIATED) || originalPayment.getStatus().equals(SUCCESS)) {
+            return claim.toBuilder()
+                .claimData(claim.getClaimData().toBuilder()
+                    .payment(originalPayment)
+                    .build())
+                .build();
+        }
+
+        LocalDate issuedOn = issueDateCalculator.calculateIssueDay(nowInLocalZone());
+        LocalDate responseDeadline = responseDeadlineCalculator.calculateResponseDeadline(issuedOn);
+
+        Claim updatedClaim = claim.toBuilder()
+            .ccdCaseId(caseId)
+            .issuedOn(issuedOn)
+            .responseDeadline(responseDeadline)
+            .channel(CITIZEN)
+            .build();
```

Comment:
I wonder if we may have a gap here? See above when the payment status is either `INITIATED` or `SUCCESS` we don't set the `ccdCaseId` not the `channel` explicitly whereas here we do...

Reply:
that's actually redundant because case id and channel are already set in the initiate payment. I'll remove them